lib/logger.rb

```diff
@@ -684,6 +730,7 @@ def add(severity, message = nil, progname = nil)
         progname = @progname
       end
     end
+    message = process_message(format_severity(severity), Time.now, progname, message)
     @logdev.write(
       format_message(format_severity(severity), Time.now, progname, message))
```

Comment:
Shouldn't we use the same time?

```suggestion
    now = Time.now
    message = process_message(format_severity(severity), now, progname, message)
    @logdev.write(
      format_message(format_severity(severity), now, progname, message))
```